src/process/namespaces.rs

```diff
@@ -12,22 +13,36 @@ use super::Process;
 impl Process {
     /// Describes namespaces to which the process with the corresponding PID belongs.
     /// Doc reference: https://man7.org/linux/man-pages/man7/namespaces.7.html
-    pub fn namespaces(&self) -> ProcResult<Vec<Namespace>> {
+    /// The namespace type is the key for the HashMap, i.e 'net', 'user', etc.
+    pub fn namespaces(&self) -> ProcResult<HashMap<OsString, Namespace>> {
         let ns = self.root.join("ns");
-        let mut namespaces = Vec::new();
+        let mut namespaces = HashMap::new();
         for entry in fs::read_dir(ns)? {
             let entry = entry?;
             let path = entry.path();
             let ns_type = entry.file_name();
             let stat = rustix::fs::statat(&rustix::fs::cwd(), &path, AtFlags::empty())
                 .map_err(|_| build_internal_error!(format!("Unable to stat {:?}", path)))?;
 
-            namespaces.push(Namespace {
-                ns_type,
-                path,
-                identifier: stat.st_ino,
-                device_id: stat.st_dev,
-            })
+            namespaces
+                .insert(
+                    ns_type.clone(),
+                    Namespace {
+                        ns_type,
+                        path,
+                        identifier: stat.st_ino,
+                        device_id: stat.st_dev,
+                    },
+                )
+                .map_or_else(
+                    || Ok(()),
+                    |n| {
+                        Err(build_internal_error!(format!(
+                            "NsType appears more than once {:?}",
+                            n.ns_type
+                        )))
+                    },
+                )?;
```

Comment:
Sorry to be a pain about this, but I think this can be written much more cleanly by just using a simple `if` statement:

```suggestion
            if namespaces
                .insert(
                    ns_type.clone(),
                    Namespace {
                        ns_type,
                        path,
                        identifier: stat.st_ino,
                        device_id: stat.st_dev,
                    },
                )
                .is_some()
            {
                return Err(build_internal_error!(format!("Duplicate namespace type {:?}", ns_type)));
            }
```

Reply:
No problem, pushed a new revision, please let me know if there is any other change you would like to have.